### PR TITLE
Updated AddLegacyActivity method name to AddLegacySource

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Updated `AddLegacyActivity` to `AddLegacySource`

* [x] `CHANGELOG.md` updated for non-trivial changes
* [x] Changes in public API reviewed

### CODE REVIEW

**Review by eddynaka — APPROVED** [review]

File: src/OpenTelemetry/Trace/TracerProviderBuilderSdk.cs [hunk 1]
```diff
@@ -132,9 +132,9 @@ internal TracerProviderBuilder AddProcessor(BaseProcessor<Activity> processor)
         /// <summary>
```
Comment (CodeBlanch) [inline src/OpenTelemetry/Trace/TracerProviderBuilderSdk.cs:132]: I think the code comments should use a plural tense. Here's an alternative:
```csharp
        /// <summary>
        /// Adds a listener for <see cref="Activity"/> objects created with the given operation name to the <see cref="TracerProviderBuilder"/>.
        /// </summary>
        /// <remarks>
        /// This is provided to capture legacy <see cref="Activity"/> objects creared without using the <see cref="ActivitySource"/> API.
        /// </remarks>
        /// <param name="tracerProviderBuilder"><see cref="TracerProviderBuilder"/> instance.</param>
        /// <param name="operationName">Operation name of the <see cref="Activity"/> objects to capture.</param>
        /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
        public static TracerProviderBuilder AddLegacyActivity(this TracerProviderBuilder tracerProviderBuilder, string operationName)
```

Comment (utpilla) [reply src/OpenTelemetry/Trace/TracerProviderBuilderSdk.cs:132]: I have updated the code comments.

File: src/OpenTelemetry/CHANGELOG.md [hunk 2]
```diff
@@ -9,16 +9,20 @@ please check the latest changes
 
 ## Unreleased
 
-* Added `ForceFlush` to `TracerProvider`. ([#1837](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1837))
+* Updated the method name of `AddLegacyActivity` to `AddLegacySource`
```
Comment (cijothomas) [inline src/OpenTelemetry/CHANGELOG.md:12]: no need of this. as this was never released.

File: src/OpenTelemetry/CHANGELOG.md [hunk 3]
```diff
@@ -9,16 +9,20 @@ please check the latest changes
...
+  [#1860](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1860)
 
-* Added a TracerProvierBuilder extension method called
-  `AddLegacyActivityOperationName` which is used by instrumentation libraries
-  that use DiagnosticSource to get activities processed without
-  ActivitySourceAdapter.
+* Added `ForceFlush` to `TracerProvider`.
+  ([#1837](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1837))
+
+* Added a TracerProviderBuilder extension method called `AddLegacyActivity`
```
Comment (cijothomas) [inline src/OpenTelemetry/CHANGELOG.md:18]: this is good enough.

Comment (utpilla) [reply src/OpenTelemetry/CHANGELOG.md:18]: Should I update the method name to `AddLegacySource` here? We would want CHANGELOG.md to show that the public API with the new name was added, right?

Comment (utpilla) [reply src/OpenTelemetry/CHANGELOG.md]: Updated the CHANGELOG.md

**Pull request merged**